.github/workflows/polonius.yml

```diff
@@ -0,0 +1,61 @@
+# Rustc by default uses the NLL borrow-checker. There are multiple cases where
+# NLL is too restrictive. An upcoming improvement to Rust's borrow-checker,
+# adressing these shortcomings, is Polonius. It ships with Nightly behind a
+# flag.
+#
+# This workflow first removes the workarounds necessary to please NLL and then
+# runs with Polonius to ensure each workaround only fixes the false-positive of
+# NLL and doesn't mask an actually undefined behavior.
```

Comment:
Somewhat unconventional. Still, I think statically proving `unsafe` code to be safe is worth it. As always, open for alternative suggestions.